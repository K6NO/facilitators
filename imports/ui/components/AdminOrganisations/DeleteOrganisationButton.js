/* eslint-disable max-len, no-return-assign */
import React from 'react';
import { Button } from 'reactstrap';
import PropTypes from 'prop-types';

// import './DeleteOrganisationButton.scss';

class DeleteOrganisationButton extends React.Component {
  constructor(props){
    super(props);
  }

  deleteOrganisation = () => {
    const { organisationId } = this.props;
    if (confirm('Delete organisation? This is permanent!')) {
        Meteor.call('organisations.remove', organisationId);
    }
  }

  render() {
    return ( 
        <Button 
          color="danger" 
          onClick={() => this.deleteOrganisation()}>
            Delete
          </Button>
    );
  }
}

DeleteOrganisationButton.propTypes = {
  organisationId: PropTypes.string.isRequired,
};

export default DeleteOrganisationButton;