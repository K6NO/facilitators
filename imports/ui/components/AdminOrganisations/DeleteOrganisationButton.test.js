import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Button } from 'reactstrap';
import DeleteOrganisationButton from './DeleteOrganisationButton';
  
const organisationId = '3454365467567fgfdgfdgfd';

test('Admin DeleteOrganisationButton with organisationId', () => {
    const component = renderer.create(
        <DeleteOrganisationButton organisationId={organisationId} />
        );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Admin delete organisation button click calls function', () => {
    const mockFunction = jest.fn();
    const component = shallow(
        <Button color="danger" onClick={mockFunction}>Delete</Button>
    );
    const button = component.find('button');
    button.simulate('click');
    expect(button.length).toBe(1)
    expect(mockFunction.mock.calls.length).toEqual(1);
});