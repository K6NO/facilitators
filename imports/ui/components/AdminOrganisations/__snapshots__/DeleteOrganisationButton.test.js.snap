// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin DeleteOrganisationButton with organisationId 1`] = `
<button
  aria-label={null}
  className="btn btn-danger"
  onClick={[Function]}
  type="button"
>
  Delete
</button>
`;
